packages/node-cli/src/commands/CreateProjectCommandGenerator.js

```diff
@@ -221,104 +224,115 @@ module.exports = class CreateProjectCommandGenerator extends BaseCommandGenerato
 	}
 
 	async _executeAction(answers) {
-		const projectFolderName = answers[COMMAND_ANSWERS.PROJECT_FOLDER_NAME];
-		const projectAbsolutePath = answers[COMMAND_OPTIONS.PARENT_DIRECTORY];
-		const manifestFilePath = path.join(
-			projectAbsolutePath,
-			SOURCE_FOLDER,
-			ApplicationConstants.FILES.MANIFEST_XML
-		);
+		try {
+			const projectFolderName = answers[COMMAND_ANSWERS.PROJECT_FOLDER_NAME];
+			const projectAbsolutePath = answers[COMMAND_OPTIONS.PARENT_DIRECTORY];
+			const manifestFilePath = path.join(
+				projectAbsolutePath,
+				SOURCE_FOLDER,
+				ApplicationConstants.FILES.MANIFEST_XML
+			);
 
-		const validationErrors = this._validateParams(answers);
+			const validationErrors = this._validateParams(answers);
 
-		if (validationErrors.length > 0) {
-			throwValidationException(validationErrors, false, this._commandMetadata);
-		}
+			if (validationErrors.length > 0) {
+				throwValidationException(validationErrors, false, this._commandMetadata);
+			}
+
+			const params = {
+				//Enclose in double quotes to also support project names with spaces
+				parentdirectory: CommandUtils.quoteString(projectAbsolutePath),
+				type: answers[COMMAND_OPTIONS.TYPE],
+				projectname: SOURCE_FOLDER,
+				...(answers[COMMAND_OPTIONS.OVERWRITE] && { overwrite: '' }),
+				...(answers[COMMAND_OPTIONS.TYPE] === ApplicationConstants.PROJECT_SUITEAPP && {
+					publisherid: answers[COMMAND_OPTIONS.PUBLISHER_ID],
+					projectid: answers[COMMAND_OPTIONS.PROJECT_ID],
+					projectversion: answers[COMMAND_OPTIONS.PROJECT_VERSION],
+				}),
```

Comment:
can we construct this in a more-readable way? like:
```
const params = {
  //Enclose in double quotes to also support project names with spaces
  parentdirectory: CommandUtils.quoteString(projectAbsolutePath),
  type: answers[COMMAND_OPTIONS.TYPE],
  projectname: SOURCE_FOLDER
}
if (answers[COMMAND_OPTIONS.OVERWRITE]) { 
  params.overwrite = ''
}
if (answers[COMMAND_OPTIONS.TYPE] === ApplicationConstants.PROJECT_SUITEAPP) {
  params.publisherid: answers[COMMAND_OPTIONS.PUBLISHER_ID],
  params.projectid: answers[COMMAND_OPTIONS.PROJECT_ID],
  params.projectversion: answers[COMMAND_OPTIONS.PROJECT_VERSION]
}
```

Reply:
yeah, I understand what you says... Using "..." is enough to express conditionality but this is already a little bit complex.
So, it's ok for me to change it but since I'm agnostic about javascript code format let's wait for the opinion of the other people.

Reply:
I have done the changes finally as suggested.